src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java

```diff
@@ -171,84 +157,92 @@ public IBlockState getStateFromMeta(final int meta)
         return this.getDefaultState().withProperty(FACING, enumfacing);
     }
 
-    /**
-     * @deprecated (Remove this as soon as minecraft offers anything better).
-     */
-    @SideOnly(Side.CLIENT)
     @Override
-    @Deprecated
-    public boolean shouldSideBeRendered(IBlockState blockState, IBlockAccess blockAccess, BlockPos pos, EnumFacing side)
+    public int getMetaFromState(@NotNull final IBlockState state)
     {
-        return true;
+        return state.getValue(FACING).getIndex();
     }
 
     /**
      * @deprecated (Remove this as soon as minecraft offers anything better).
      */
+    @SuppressWarnings(DEPRECATION)
     @Override
     @Deprecated
-    public boolean isOpaqueCube(final IBlockState state)
+    public boolean isFullCube(final IBlockState state)
     {
         return false;
     }
 
-    @NotNull
     @Override
-    protected BlockStateContainer createBlockState()
+    public boolean isPassable(final IBlockAccess worldIn, final BlockPos pos)
     {
-        return new BlockStateContainer(this, FACING);
+        return true;
     }
 
-    @NotNull
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @SuppressWarnings(DEPRECATION)
     @Override
-    @SideOnly(Side.CLIENT)
-    public BlockRenderLayer getBlockLayer()
+    @Deprecated
+    public AxisAlignedBB getBoundingBox(final IBlockState state, final IBlockAccess source, final BlockPos pos)
     {
-        return BlockRenderLayer.SOLID;
+
+        if (state.getValue(FACING).equals(EAST) || state.getValue(FACING).equals(WEST))
+        {
+            return new AxisAlignedBB((float) WE_START_COLLISION_X,
+                    (float) BOTTOM_COLLISION,
+                    (float) WE_START_COLLISION_Z,
+                    (float) WE_END_COLLISION_X,
+                    (float) HEIGHT_COLLISION,
+                    (float) WE_END_COLLISION_Z);
+        }
+        else
+        {
+            return new AxisAlignedBB((float) SN_START_COLLISION_X,
+                    (float) BOTTOM_COLLISION,
+                    (float) SN_START_COLLISION_Z,
+                    (float) SN_END_COLLISION_X,
+                    (float) HEIGHT_COLLISION,
+                    (float) SN_END_COLLISION_Z);
+        }
     }
 
     /**
      * @deprecated (Remove this as soon as minecraft offers anything better).
      */
+    @SuppressWarnings(DEPRECATION)
+    @SideOnly(Side.CLIENT)
     @Override
     @Deprecated
-    public boolean isFullCube(final IBlockState state)
+    public boolean shouldSideBeRendered(final IBlockState blockState, final IBlockAccess blockAccess, final BlockPos pos, final EnumFacing side)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)